src/main/java/com/minecolonies/colony/buildings/BuildingGuardTower.java

```diff
@@ -0,0 +1,187 @@
+package com.minecolonies.colony.buildings;
+
+import com.minecolonies.client.gui.WindowHutWorkerPlaceholder;
+import com.minecolonies.colony.CitizenData;
+import com.minecolonies.colony.Colony;
+import com.minecolonies.colony.ColonyView;
+import com.minecolonies.colony.jobs.AbstractJob;
+import com.minecolonies.colony.jobs.JobGuard;
+import net.minecraft.entity.SharedMonsterAttributes;
+import net.minecraft.util.math.BlockPos;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Building class of the guard tower.
+ */
+public class BuildingGuardTower extends AbstractBuildingWorker
+{
+    /**
+     * Name description of the guard hat.
+     */
+    private static final String GUARD_TOWER = "GuardTower";
+
+    /**
+     * Max level of the guard hut.
+     */
+    private static final int GUARD_HUT_MAX_LEVEL = 5;
+
+    /**
+     * The max vision bonus multiplier
+     */
+    private static final int MAX_VISION_BONUS_MULTIPLIER = 3;
+
+    /**
+     * The health multiplier each level after level 4.
+     */
+    private static final int HEALTH_MULTIPLIER = 2;
+
+    /**
+     * Base max health of the guard.
+     */
+    private static final double BASE_MAX_HEALTH = 20D;
+
+    /**
+     * Vision bonus per level.
+     */
+    private static final int VISION_BONUS = 5;
+
+    /**
+     * Constructor for the guardTower building.
+     *
+     * @param c Colony the building is in.
+     * @param l Location of the building.
+     */
+    public BuildingGuardTower(Colony c, BlockPos l)
+    {
+        super(c, l);
+    }
+
+    /**
+     * Gets the name of the schematic.
+     *
+     * @return Baker schematic name.
+     */
+    @NotNull
+    @Override
+    public String getSchematicName()
+    {
+        return GUARD_TOWER;
+    }
+
+    /**
+     * Gets the max level of the baker's hut.
+     *
+     * @return The max level of the baker's hut.
+     */
+    @Override
+    public int getMaxBuildingLevel()
+    {
+        return GUARD_HUT_MAX_LEVEL;
+    }
+
+    /**
+     * The name of the baker's job.
+     *
+     * @return The name of the baker's job.
+     */
+    @NotNull
+    @Override
+    public String getJobName()
+    {
+        return GUARD_TOWER;
+    }
+
+    /**
+     * If no vision multiplier give health bonus.
+     * @return the bonus health.
+     */
+    public int getBonusHealth()
+    {
+        if (getBuildingLevel() > MAX_VISION_BONUS_MULTIPLIER)
+        {
+            return (getBuildingLevel() - MAX_VISION_BONUS_MULTIPLIER) * HEALTH_MULTIPLIER;
+        }
+        return 0;
+    }
+
+    @Override
+    public void onUpgradeComplete(final int newLevel)
```

Comment:
Can we make that in a seperate pr?
I feel like we're overloading prs with additional stuff a lot.
Would also make a great task for a new one in our team.